### PR TITLE
App v2

### LINKED ISSUE #3: Don't require a / in the beginning
It would be nice if we could simply write the following anywhere in a comment: `Depends on #something, #something, and #something` and it would work. Writing a slash command in a comment feels weird.

### LINKED ISSUE #8: Allow commands also within PR description
Currently the bot only recognizes the `/depends` command in a PR comment. It would be nice if the bot would also recognize the command within the PR description.

### PR DESCRIPTION
**Pull request by z0al**
This PR deprecates the `/depends` and `/ensure` commands in favor of the new single inline command: `depend` (no slash). The new command works similar to how you may [close issues using keywords][ref] on GitHub but **only** in pull request description.

The following keywords followed by an issue number will mark that issue as a dependency:

* depend on
* depends on
* depended on

### Why new syntax?

* Easier to read, add, update, and remove dependencies (e.g: You can immediately tell what are the current dependencies without looking at comments).
* Handles permissions by default (no need to check the permissions of the commenter, thanks to @ryanhiebert)
* It matches the way GitHub works.
* Removes the need for workarounds (e.g. probot-metadata)
* It has been requested on #3 and #8.

Why not support both? because I think things will get messy really soon, and the source would be harder to maintain!

### Examples

You may write the end of PR description:

> Depends on #&#8203;1

or it can be anywhere:

> This pull request **depends on #&#8203;1** which does bla bla bla. Oh, and it **depends on #&#8203;2** too.

All dependencies added using the old commands should be converted automatically (hopefully) to the new syntax. The old commands will just write a deprecation notice.

/CC for review and comments @nesl247, @ryanhiebert @lo1tuma, and @JasonEtco ;)

Resolves #3 and resolves #8

[ref]: https://git.io/vNxHZ

### CODE REVIEW

File: lib/helpers/deprecate.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+-------
+
+🤖 This comment was generated by [**DEP [bot]**][repo]. Please report issues \
+[here][issues].
+
+Happy coding!
+
+[ref]: https://git.io/vNxHZ
+[repo]: https://github.com/ahmed-taj/dep
+[issues]: https://github.com/ahmed-taj/dep/issues
```
Comment (ryanhiebert) [inline lib/helpers/deprecate.js:41]: very cool!

**Review by ryanhiebert — APPROVED** [review]
Looks pretty good. I didn't run it or anything, but I see a lot that I like here!

**Comment by z0al** [comment]
This PR will be merged into the master branch (=> immediate deploy) on 15 Feb if no issues found. Thanks!

**Comment by ahmed-taj** [comment]
I'm currently having trouble with [Heroku](https://heroku.com). The app crashed :/

**Pull request merged**